src/otx/algo/detection/yolov9.py

```diff
@@ -0,0 +1,196 @@
+        super().__init__(
+            model_name=model_name,
+            label_info=label_info,
+            input_size=input_size,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            metric=metric,
+            torch_compile=torch_compile,
+            tile_config=tile_config,
+        )
+
+    def _build_model(self, num_classes: int) -> SingleStageDetector:
+        backbone = GELAN(model_name=self.model_name)
+        neck = YOLONeck(model_name=self.model_name)
+        bbox_head = YOLOHead(model_name=self.model_name, num_classes=num_classes)
+
+        # patch _load_from_state_dict
+        SingleStageDetector._load_from_state_dict = _load_from_state_dict_for_yolov9  # type: ignore[method-assign] # noqa: SLF001
+        detector = SingleStageDetector(
+            backbone=backbone,
+            neck=neck,
+            bbox_head=bbox_head,
+            criterion=None,
+        )
+
+        # set criterion
+        strides: list[int] | None = [8, 16, 32] if self.model_name == "yolov9_c" else None
+        self.vec2box = Vec2Box(detector, self.input_size, strides)
+        detector.bbox_head.vec2box = self.vec2box
+        detector.criterion = YOLOv9Criterion(
+            num_classes=num_classes,
+            loss_cls=BCELoss(),
+            loss_dfl=DFLoss(self.vec2box),
+            loss_iou=BoxLoss(),
+            vec2box=self.vec2box,
+        )
+
+        return detector
+
+    @property
+    def _exporter(self) -> OTXModelExporter:
+        """Creates OTXModelExporter object that can export the model."""
+        if self.input_size is None:
+            msg = f"Input size attribute is not set for {self.__class__}"
+            raise ValueError(msg)
+
+        if self.explain_mode:
+            msg = "Explainable model export is not supported for YOLOv9 yet."
+            raise NotImplementedError(msg)
+
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=(1, 3, *self.input_size),
+            mean=self.mean,
+            std=self.std,
+            resize_mode="fit_to_window_letterbox",
+            pad_value=114,
+            swap_rgb=True,
+            via_onnx=True,
+            onnx_export_configuration={
+                "input_names": ["image"],
+                "output_names": ["boxes", "labels"],
+                "export_params": True,
+                "opset_version": 11,
+                "dynamic_axes": {
+                    "image": {0: "batch", 2: "height", 3: "width"},
+                    "boxes": {0: "batch", 1: "num_dets"},
+                    "labels": {0: "batch", 1: "num_dets"},
+                },
+                "keep_initializers_as_inputs": False,
+                "verbose": False,
+                "autograd_inlining": False,
+            },
+            output_names=None,  # TODO (someone): support XAI
+        )
+
+    def to(self, *args, **kwargs) -> Self:
```

Comment:
It's just a question. What is it for?

Reply:
Some modules related to `Vec2Box` are not converted to the target device.
I'm not sure, but I'm guessing that it is because `Vec2Box` must be created after model instanstiation because auto anchor generation in the module is based on model forward's output.
Here is to manually convert the modules.